lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -44,15 +44,17 @@
 @DriverSide
 @Private
 @Unstable
-public final class DriverRestartManager implements DriverIdlenessSource {
+public final class DriverRestartManager implements DriverIdlenessSource, AutoCloseable {
+
   private static final String CLASS_NAME = DriverRestartManager.class.getName();
   private static final Logger LOG = Logger.getLogger(CLASS_NAME);
 
+  private final Timer restartCompletedTimer = new Timer(this.getClass().getSimpleName() + ":Timer");
```

Comment:
@markusweimer yes - I do not really know all details of the restart logic, but the timer is a thread that prevents us from clean shutdown when the REEF app is closed.